### DISCUSSION TITLE
file import problem

### DISCUSSION THREAD

**Discussion by mouhamadou-ndim** [Toblerity/Fiona#1108] [Q&A] [answered]
Hello,
I am trying to import my files which are in .gpkg format. 
rpg15 = gpd.read_file(path_rpg15)
rpg16 = gpd.read_file(path_rpg16)
rpg17 = gpd.read_file(path_rpg17)
rpg18 = gpd.read_file(path_rpg18)
rpg19 = gpd.read_file(path_rpg19)

But I always get an error message like this:

CPLE_OpenFailedError                      Traceback (most recent call last)
File fiona\_shim.pyx:83, in fiona._shim.gdal_open_vector()

File fiona\_err.pyx:291, in fiona._err.exc_wrap_pointer()

CPLE_OpenFailedError: rpg_15.gpkg: No such file or directory

During handling of the above exception, another exception occurred:

DriverError                               Traceback (most recent call last)
Input In [17], in <cell line: 2>()
      1 # import des RPG
----> 2 rpg15 = gpd.read_file(path_rpg15)
      3 rpg16 = gpd.read_file(path_rpg16)
      4 rpg17 = gpd.read_file(path_rpg17)

File ~\Anaconda3\envs\geo_env\lib\site-packages\geopandas\io\file.py:201, in _read_file(filename, bbox, mask, rows, **kwargs)
    198     reader = fiona.open
    200 with fiona_env():
--> 201     with reader(path_or_bytes, **kwargs) as features:
    202 
    203         # In a future Fiona release the crs attribute of features will
    204         # no longer be a dict, but will behave like a dict. So this should
    205         # be forwards compatible
    206         crs = (
    207             features.crs["init"]
    208             if features.crs and "init" in features.crs
    209             else features.crs_wkt
    210         )
    212         # handle loading the bounding box

File ~\Anaconda3\envs\geo_env\lib\site-packages\fiona\env.py:408, in ensure_env_with_credentials.<locals>.wrapper(*args, **kwargs)
    405 @wraps(f)
    406 def wrapper(*args, **kwargs):
    407     if local._env:
--> 408         return f(*args, **kwargs)
    409     else:
    410         if isinstance(args[0], str):

File ~\Anaconda3\envs\geo_env\lib\site-packages\fiona\__init__.py:264, in open(fp, mode, driver, schema, crs, encoding, layer, vfs, enabled_drivers, crs_wkt, **kwargs)
    261     path = parse_path(fp)
    263 if mode in ('a', 'r'):
--> 264     c = Collection(path, mode, driver=driver, encoding=encoding,
    265                    layer=layer, enabled_drivers=enabled_drivers, **kwargs)
    266 elif mode == 'w':
    267     if schema:
    268         # Make an ordered dict of schema properties.

File ~\Anaconda3\envs\geo_env\lib\site-packages\fiona\collection.py:162, in Collection.__init__(self, path, mode, driver, schema, crs, encoding, layer, vsi, archive, enabled_drivers, crs_wkt, ignore_fields, ignore_geometry, **kwargs)
    160 if self.mode == 'r':
    161     self.session = Session()
--> 162     self.session.start(self, **kwargs)
    163 elif self.mode in ('a', 'w'):
    164     self.session = WritingSession()

File fiona\ogrext.pyx:540, in fiona.ogrext.Session.start()

File fiona\_shim.pyx:90, in fiona._shim.gdal_open_vector()

DriverError: rpg_15.gpkg: No such file or directory

It seems to me that the fiona package is installed. Can you help me please?

**Comment by rbuffat** [comment, marked as answer]
@Mhdx20 

> DriverError: rpg_15.gpkg: No such file or directory

It looks as if the file you want to open does not exist. You can check if a file exists with `os.path.exists(path_rpg15)`.

**Reply by mouhamadou-ndim** [reply]
@rbuffat 
Yes the file exists in my directory, but I don't understand why it can't find it

**Reply by rbuffat** [reply]
@Mhdx20 If you use a relative path and your current working directory is not the directory with the data files, the file will not be found. You could either use absolute paths or ensure that your current working directory corresponds to the directory with the data files.

**Reply by mouhamadou-ndim** [reply]
@rbuffat  Thank you very much, it was indeed related to a directory problem, it's fixed.

**Reply by mouhamadou-ndim** [reply]
I have an error on my screen. I want to create a single geometry column with the following lines of code:
rpg15_19.loc[rpg15_19.geometry_x != None, 'geom'] = rpg15_19['geometry_x']
rpg15_19.loc[rpg15_19.geometry_y != None, 'geom'] = rpg15_19['geometry_y']
rpg15_19.loc[rpg15_19.geometry != None, 'geom'] = rpg15_19['geometry']

I get an error message like this:
KeyError Traceback (most recent call last)
Input In [7], in <cell line: 2>()
      1 # only one geometry column is created
----> 2 rpg15_19.loc[rpg15_19.geometry_x != None, 'geom'] = rpg15_19['geometry_x']
      3 rpg15_19.loc[rpg15_19.geometry_y != None, 'geom'] = rpg15_19['geometry_y']
      4 rpg15_19.loc[rpg15_19.geometry != None, 'geom'] = rpg15_19['geometry']

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\coreindexing.py:712, in _LocationIndexer.__setitem__(self, key, value)
    710 else:
    711 key = com.apply_if_callable(key, self.obj)
--> 712 indexer = self._get_setitem_indexer(key)
    713 self._has_valid_setitem_indexer(key)
    715 iloc = self if self.name == "iloc" else self.obj.iloc

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:661, in _LocationIndexer._get_setitem_indexer(self, key)
    659 if isinstance(key, tuple):
    660 with suppress(IndexingError):
--> 661 return self._convert_tuple(key)
    663 if isinstance(key, range):
    664 return list(key)

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\coreindexing.py:799, in _LocationIndexer._convert_tuple(self, key)
    797 self._validate_key_length(key)
    798 for i, k in enumerate(key):
--> 799 idx = self._convert_to_indexer(k, axis=i)
    800 keyidx.append(idx)
    802 return tuple(keyidx)

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:1291, in _LocIndexer._convert_to_indexer(self, key, axis)
   1289 return inds
   1290 else:
-> 1291 return self._get_listlike_indexer(key, axis)[1]
   1292 else:
   1293 try:

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:1327, in _LocIndexer._get_listlike_indexer(self, key, axis)
   1324 ax = self.obj._get_axis(axis)
   1325 axis_name = self.obj._get_axis_name(axis)
-> 1327 keyarr, indexer = ax._get_indexer_strict(key, axis_name)
   1329 return keyarr, indexer

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\coreindexes\base.py:5782, in Index._get_indexer_strict(self, key, axis_name)
   5779 else:
   5780 keyarr, indexer, new_indexer = self._reindex_non_unique(keyarr)
-> 5782 self._raise_if_missing(keyarr, indexer, axis_name)
   5784 keyarr = self.take(indexer)
   5785 if isinstance(key, Index):
   5786 # GH 42790 - Preserve name from an Index

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\coreindexes\base.py:5842, in Index._raise_if_missing(self, key, indexer, axis_name)
   5840 if use_interval_msg:
   5841 key = list(key)
-> 5842 raise KeyError(f "None of [{key}] are in the [{axis_name}]")
   5844 not_found = list(ensure_index(key)[missing_mask.nonzero()[0]].unique())
   5845 raise KeyError(f"{not_found} not in index")

KeyError: "None of [Index([('g', 'e', 'o', 'm', 'e', 't', 'r', 'y', '_', 'x'), ('g', 'e', 'o', 'm', 'e', 't', 'r', 'y', '_', 'x')], dtype='object')] are in the [index]"
